core/interfaces/i_autohideable.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a component that is automatically hidden
+ * when Blockly.hideChaff is called.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IAutoHideable');
+
+
+goog.require('Blockly.IPlugin');
+
+
+/**
+ * Interface for a component that can be automatically hidden.
+ * @extends {Blockly.IPlugin}
+ * @interface
+ */
+Blockly.IAutoHideable = function() {};
+
+/**
+ * Hides the component. Called in Blockly.hideChaff.
```

Comment:
Similarly I think this could use some more documentation, about how to use/register it, and what the implementer should do in this method. But I think if you wanted to do that separately, that's fine too since perhaps the PluginManager api is changing.

Reply:
I expect to add documentation, probably during bugbash, when adding documentation for the `PluginManager`.